php-checks/src/main/java/org/sonar/php/checks/HardCodedIpAddressCheck.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.expression.LiteralTree;
+import org.sonar.plugins.php.api.visitors.PHPVisitorCheck;
+
+@Rule(key = "S1313")
+public class HardCodedIpAddressCheck extends PHPVisitorCheck {
+
+  private static final Matcher IP = Pattern.compile("([^\\d.]*\\/)?((?<ip>(?:\\d{1,3}\\.){3}\\d{1,3})(:\\d{1,5})?(?!\\d|\\.))(\\/.*)?").matcher("");
```

Comment:
What about IPv6?

Reply:
I was looking at IPv6 and it seems to be quite complicated, but perhaps I can cover some simpler cases

Reply:
We can put IPv6 aside for the moment and create a ticket so that we don't forget it.

Reply:
I created https://jira.sonarsource.com/browse/SONARPHP-812